saleor/core/utils/__init__.py

```diff
@@ -111,3 +118,37 @@ def create_thumbnails(pk, model, size_set, image_attr=None):
         logger.info("Created %d thumbnails", num_created)
     if failed_to_create:
         logger.error("Failed to generate thumbnails", extra={"paths": failed_to_create})
+
+
+def generate_unique_slug(
+    instance: Type[Model], slugable_value: str, slug_field_name: str = "slug",
+) -> str:
+    """Create unique slug for model instance.
```

Comment:
Maby some hints with an example of how we generate this unique slug. Good to know what is appended to slug without reading the whole function.

Reply:
You thinking about something like this in docstring:
```
    Example:
        For Product model where product variable is a Product instance:
        >>> generate_unique_slug(product, product.name, "slug")
```

Reply:
I think @fowczarek meant that we could roughly explain the logic behind generating unique slugs. Something like:

```
"""
Create unique slug for model instance.

The function uses `django.utils.text.slugify` to generate a slug from the `slugable_value` model field. If the slug already exists it adds a numeric suffix and increments it until a unique value is found.
"""
```